### PR TITLE
load-aware: dump trimaran log during deploy

### PR DESCRIPTION
**Pull request by drewrip**
This should fix the suggestions in #855 and #863.

### CODE REVIEW

**Comment by drewrip** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/877/pull-ci-openshift-psap-ci-artifacts-main-load-aware-light/1684943025744121856/artifacts/light/test/artifacts/001__default_plots/_matbench_visualize.log). The plots and reports were successfully generated with only 1 data point.

@jmencak, this should be good to merge whenever you have a minute. Not an urgent change though.

File: roles/load_aware/load_aware_deploy_trimaran/tasks/main.yml [hunk 1]
```diff
@@ -110,6 +110,11 @@
       oc describe pod trimaran-test -n trimaran > "{{ artifact_extra_logs_dir }}/trimaran-test.pod.desc"
       oc get all -n trimaran -oyaml > "{{ artifact_extra_logs_dir }}/all-trimaran.yaml"
 
+  - name: Dump trimaran scheduler log
+    shell: |
+      oc logs -n trimaran $(oc get pod -n trimaran -l "app=trimaran-scheduler" | awk 'NR > 1 {print $1}') > "{{ artifact_extra_logs_dir }}/trimaran_scheduler.log"
```
Comment (jmencak) [inline roles/load_aware/load_aware_deploy_trimaran/tasks/main.yml:115]: Let's make this a bit cleaner and use `--no-headers` when listing the pods.  Then you'll not need to use 'NR > 1'.

File: visualizations/load-aware/plotting/comparison.py [hunk 2]
```diff
@@ -86,9 +86,13 @@ def ExecutionTimeComparison_generatePodTimes(entry, variables, exclude_workload=
 
 def ExecutionTimeComparison_generateStats(entry, variables, data):
     df = pd.DataFrame(data)
-    q1, med, q3 = stats.quantiles(df.Duration)
-    q90 = stats.quantiles(df.Duration, n=10)[8] # 90th percentile
-    q100 = df.Duration.max()
+    q1, med, q3 = (0.0, 0.0, 0.0)
+    q90 = 0.0
+    q100 = 0.0
+    if len(df) >= 2:
```
Comment (jmencak) [inline visualizations/load-aware/plotting/comparison.py:92]: I didn't look into the data model, so I'm not sure why we need len(df) >=2 but I'm assuming you know what you're doing.  Similarly below.

**Review by jmencak — COMMENTED** [review]
The changes look sane to me. Minor nit and I'm not sure about the len >= 2.  If I have some explanation, happy to approve.

File: visualizations/load-aware/plotting/comparison.py (hunk 2 above)
Comment (drewrip) [reply visualizations/load-aware/plotting/comparison.py:92]: Having less than 2 data points causes the stats functions to panic, so we avoid calling them if this is the case.

[Some previous discussion](https://github.com/openshift-psap/ci-artifacts/pull/863#issuecomment-1653385656)

File: roles/load_aware/load_aware_deploy_trimaran/tasks/main.yml (hunk 1 above)
Comment (drewrip) [reply roles/load_aware/load_aware_deploy_trimaran/tasks/main.yml:115]: Will do!

**Comment by jmencak** [comment]
Thank you for the changes and explanation. LGTM, let's merge once the tests succeed.

**Comment by drewrip** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/877/pull-ci-openshift-psap-ci-artifacts-main-load-aware-light/1686017185241108480/artifacts/light/test/artifacts/001__default_plots/_matbench_visualize.log). Should be good to merge.

**Pull request merged**